scipy/stats/_distribution_infrastructure.py

```diff
@@ -0,0 +1,5082 @@
+        the resulting probability is very close to `0` or `1` - too close to
+        represent accurately with floating point arithmetic. In many cases,
+        however, the *logarithm* of this resulting probability may be
+        represented in floating point arithmetic, in which case this function
+        may be used to find the argument of the CCDF for which the *logarithm*
+        of the resulting probability is `x`.
+
+        Due to the inherent spacing between floating point numbers, it is common
+        for there to be no value of the argument for which the log-CCDF is close
+        to the provided `x`. In such cases, this function returns NaN.
+
+        See Also
+        --------
+        ContinuousDistribution.iccdf
+        ContinuousDistribution.ilogccdf
+
+        Examples
+        --------
+        Instantiate a distribution with the desired parameters:
+
+        >>> import numpy as np
+        >>> from scipy import stats
+        >>> X = stats.Uniform(a=-0.5, b=0.5)
+
+        Evaluate the inverse log-CCDF at the desired argument:
+
+        >>> X.ilogccdf(-0.25)
+        -0.2788007830714034
+        >>> np.allclose(X.ilogccdf(-0.25), X.iccdf(np.exp(-0.25)))
+        True
+
+        This function may return NaN when the spacing between floating point
+        values is too coarse to represent an argument for which the log-CCDF is
+        close to the provided `x`, even when the function is theoretically
+        continuous within the domain of the argument.
+
+        >>> X.ilogccdf(-np.inf), X.ilogccdf(-1e10), X.ilogccdf(-30.)
+        (0.5, nan, 0.4999999999998863)
+
+        """
+        return self._ilogccdf_dispatch(x, method=method, **self._parameters)
+
+    @_dispatch
+    def _ilogccdf_dispatch(self, x, method=None, **kwargs):
+        if self._overrides('_ilogccdf_formula'):
+            method = self._ilogccdf_formula
+        elif self._overrides('_ilogcdf_formula'):
+            method = self._ilogccdf_complementarity
+        else:
+            method = self._ilogccdf_inversion
+        return method
+
+    def _ilogccdf_formula(self, x, **kwargs):
+        return NotImplementedError(self._not_implemented)
+
+    def _ilogccdf_complementarity(self, x, **kwargs):
+        return self._ilogcdf_dispatch(_log1mexp(x), **kwargs)
+
+    def _ilogccdf_inversion(self, x, **kwargs):
+        return self._solve_bounded(self._logccdf_dispatch, x, kwargs=kwargs)
+
+    @_set_invalid_nan
+    def iccdf(self, x, *, method=None):
+        r"""Inverse complementary cumulative distribution function
+
+        The inverse complementary cumulative distribution function
+        :math:`G^{-1}(x)` is the argument :math:`y` for which the
+        complementary cumulative distribution function :math:`G(y)`
+        evaluates to :math:`x`.
+
+        .. math::
+
+            G^{-1}(x) = y \text{ such that } G(y) = x`
+
+        ``iccdf`` accepts `x` for :math:`x \in [0, 1]`.
```

Comment:
Mentioned this elsewhere, adding here to visibility: consider standardizing on `q = ccdf(x)`, `x = iccdf(q)`

Reply:
I won't miss https://github.com/mdhaber/scipy/pull/110#discussion_r1551356979 unless someone else resolves it. Resolving this to merge with that.